packages/astro/src/runtime/server/render/page.ts

```diff
@@ -131,7 +147,68 @@ export async function renderPage(
 		result.componentMetadata.get(componentFactory.moduleId!)?.containsHead ?? false;
 	const factoryReturnValue = await componentFactory(result, props, children);
 	const factoryIsHeadAndContent = isHeadAndContent(factoryReturnValue);
-	if (isRenderTemplateResult(factoryReturnValue) || factoryIsHeadAndContent) {
+	if (result._metadata.request.headers.get('x-astro-outlet') && (isRenderTemplateResult(factoryReturnValue)  || factoryIsHeadAndContent)) {
+		const outlet = result._metadata.request.headers.get('x-astro-outlet')!;
+		result.scripts.clear();
+		await bufferSlottedContent(result, outlet);
```

Comment:
If we get a request for a single `Outlet`, we should partially render the page until we find a match, then respond with just the content for that `Outlet`.